app/src/main/java/com/gh4a/service/RepositoryService.java

```diff
@@ -0,0 +1,153 @@
+import java.util.HashMap;
+import java.util.Locale;
+import io.reactivex.Observable;
+
+public class RepositoryService {
+    public static final int LOAD_README = 0;
+    public static final int LOAD_PULL_REQUESTS_COUNT = 1;
+    public static final int UPDATE_STAR = 2;
+    public static final int UPDATE_WATCH = 3;
+    public static final int IS_STARRING = 4;
+    public static final int IS_WATCHING = 5;
+    public static final int LOAD_REPOSITORY = 6;
+
+    public static Observable loadReadme(Activity activity, String repoOwner, String repoName, String ref) {
+        Gh4Application app = (Gh4Application) activity.getApplicationContext();
+
+        return Observable.create(emitter -> {
+            Log.d("TEST", "loading Readme file from wweb NETWORK CALL");
+            GitHubClient client = new DefaultClient("application/vnd.github.v3.html");
+            client.setOAuth2Token(app.getAuthToken());
+
+            ContentsService contentService = new ContentsService(client);
+            try {
+                String html = contentService.getReadmeHtml(new RepositoryId(repoOwner, repoName), ref);
+                if (html != null) {
+                    emitter.onNext(HtmlUtils.rewriteRelativeUrls(html, repoOwner, repoName, ref));
+                    emitter.onComplete();
+                }
+            } catch (RequestException e) {
+                /* don't spam logcat with 404 errors, those are normal */
+                if (e.getStatus() != 404) {
+                    emitter.onError(e);
+                }
+            } catch (IOException ioe) {
+                emitter.onError(ioe);
+            }
+        });
+    }
+
+    public static Observable loadPullRequestCount(Activity activity, Repository repository, String state) {
+        return Observable.create(emitter -> {
+            final String QUERY_FORMAT = "type:pr repo:%s/%s state:%s";
+            IssueService issueService = (IssueService)
+                    Gh4Application.get().getService(Gh4Application.ISSUE_SERVICE);
+            HashMap<String, String> filterData = new HashMap<>();
+            filterData.put("q", String.format(Locale.US, QUERY_FORMAT,
+                    repository.getOwner().getLogin(), repository.getName(), state));
+            try {
+                int count = issueService.getSearchIssueResultCount(filterData);
+                emitter.onNext(count);
+                emitter.onComplete();
+            } catch(IOException ioe) {
+                emitter.onError(ioe);
+            }
+        });
+    }
+
+    public static Observable updateStar(Gh4Application app, Activity activity, String repoOwner, String repoName, boolean isStarring) {
+        return Observable.create(e -> {
+            StarService starService = (StarService)
+                    Gh4Application.get().getService(Gh4Application.STAR_SERVICE);
+            RepositoryId repoId = new RepositoryId(repoOwner, repoName);
+            if (isStarring) starService.unstar(repoId);
```

Comment:
Code style.

Reply:
I think you meant 

```java
if (isStarring) {
  starService.unstar(repoId);
} else {
  starService.star(repoId);
}
```

I don't really see how would that make code clearer ... but it's done ;)